### ISSUE TITLE
Test coverage

### ISSUE THREAD

**Issue by kjczarne** [kjczarne/metaframe#11]
As discussed, we want to use [Coverage.py](https://coverage.readthedocs.io/en/7.4.4/) to obtain the current test coverage on the repository, using that we will inform ourselves on what other unit tests still need to be implemented.

As part of this issue I would like to get the HTML report produced by `coverage` tool to look through the methods that still need tests. Once that is done I will open a new ticket for the unit tests.

**Comment by kjczarne** [comment]
@Notaklutz you might also talk to @quangbui04 as automated coverage reporting would be a very good thing to integrate into the CI pipeline as a part of issue #9
